Automattic/wp-calypso
# Launchpad - Use launchpad endpoint to retrieve latest data

Related to https://github.com/Automattic/wp-calypso/issues/73830

## Testing / Review

Testing: Medium
Review: Medium

## Proposed Changes

* Use @mattwiebe's launchpad endpoint hook (https://github.com/Automattic/wp-calypso/pull/73833) to retrieve the latest launchpad data (site_intent, checklist task statuses, and launchpad_screen) on the Launchpad screen.

## Testing Instructions



* Checkout this branch
* Create a launchpad-enabled site or use an existing site  (Test on an atomic site as well)
* Verify that you are not redirected to `/home` if launchpad is enabled (site option `launchpad_screen` = `full)
* Verify the launchpad checklist statuses are updating properly after completing a task
* Verify the test cases are passing by executing: `yarn run test-client client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/test`
* Verify you are redirected to home if the `launchpad_screen` site option is `off` or false

## Pre-merge Checklist
